internal/wasm/jit/jit_arm64.go

```diff
@@ -900,20 +906,171 @@ func (c *arm64Compiler) assignBranchTarget(labelKey string, br *obj.Prog) {
+	// If the value exceeds the length, we will branch into the default target (corresponding to len(o.Targets) index).
+	brDefaultIndex := c.compilelBranchInstruction(arm64.ABLO)
+	c.compileRegisterToRegisterInstruction(arm64.AMOVW, tmpReg, index.register)
+	c.setBranchTargetOnNext(brDefaultIndex)
+
+	// We prepare the static data which holds the offset of
+	// each target's first instruction (incl. default)
+	// relative to the beginning of label tables.
+	//
+	// For example, if we have targets=[L0, L1] and default=L_DEFAULT,
+	// we emit the the code like this at [Emit the code for each targets and default branch] below.
+	//
+	// L0:
+	//  0x123001: XXXX, ...
+	//  .....
+	// L1:
+	//  0x123005: YYY, ...
+	//  .....
+	// L_DEFAULT:
+	//  0x123009: ZZZ, ...
+	//
+	// then offsetData becomes like [0x0, 0x5, 0x8].
+	// By using this offset list, we could jump into the label for the index by
+	// "jmp offsetData[index]+0x123001" and "0x123001" can be acquired by "LEA"
+	// instruction.
+	//
+	// Note: We store each offset of 32-bite unsigned integer as 4 consecutive bytes. So more precisely,
+	// the above example's offsetData would be [0x0, 0x0, 0x0, 0x0, 0x5, 0x0, 0x0, 0x0, 0x8, 0x0, 0x0, 0x0].
+	//
+	// Note: this is similar to how GCC implements Switch statements in C.
+	offsetData := make([]byte, 4*(len(o.Targets)+1))
+	c.addStaticData(offsetData)
+
+	// "tmpReg = &offsetData[0]"
+	c.compileConstToRegisterInstruction(
+		arm64.AMOVD,
+		int64(uintptr(unsafe.Pointer(&offsetData[0]))),
+		tmpReg,
+	)
+
+	// "index.register = tmpReg + (index.register << 2) (== &offsetData[offset])"
+	c.compileAddInstructionWithLeftShiftedRegister(index.register, 2, tmpReg, index.register)
+
+	// "index.regsetr = *index.reigier (== offsetData[offset])"
+	c.compileMemoryToRegisterInstruction(arm64.AMOVW, index.register, 0, index.register)
+
+	// Now we read the address of the beginning of the jump table.
+	// In the above example, this corresponds to reading the address of 0x123001.
+	c.compileReadInstructionAddress(obj.AJMP, tmpReg)
+
+	// Now we have the address of L0 in tmp register, and the offset to the target label in the index.register.
+	// So we could achieve the br_table jump by adding them and jump into the resulting address.
+	c.compileRegisterToRegisterInstruction(arm64.AADD, tmpReg, index.register)
+
+	c.compileUnconditionalBranchToAddressOnRegister(index.register)
+
+	// We no longer need the index's register, so mark it unused.
+	c.markRegisterUnused(index.register)
+
+	// [Emit the code for each targets and default branch]
+	labelInitialInstructions := make([]*obj.Prog, len(o.Targets)+1)
+	saved := c.locationStack
+	for i := range labelInitialInstructions {
+		// Emit the initial instruction of each target.
+		init := c.newProg()
+		// We use NOP as we don't yet know the next instruction in each label.
+		// Assembler would optimize out this NOP during code generation, so this is harmless.
```

Comment:
we don't have a type called `Assembler` so cite an example of some codegen that optimizes it out